glints-dev/glints-aries
# Feature/automate publishing npm

# What changed
**Issue**: https://github.com/glints-dev/glints-aries/issues/242

This PR is to enhance `npm version` command to automate the publish process of `glints-aries` as follows:
1. Building
   - `npm run build`
2. Linting
   - `npm run lint`
   - converted `babel-eslint` to `typescript-eslint-parser` to support TypeScript
3. Testing
   - `npm run test`
4. Check if you are logged in to avoid publishing failed
   - `npm whoami`
5. Bump minor version and commit message
   -  By default, `npm version` automatically adds a git tag and commits a message for a new version.
   -  change the version format from `4.0.0-beta.66` to `4.0.66` to support automatic version bump
6. Publish to npm
   - trigger `npm postpublish`
7. Add git tag
   - `git push --tags`

Btw, `npm version` accepts three parameters `patch` | `minor` | `major`, we can rely on this rule https://docs.npmjs.com/about-semantic-versioning#incrementing-semantic-versions-in-published-packages to determine an appropriate version number.
And, If there is a publishment without bumping version, we can just use `npm publish` instead of `npm version`.
